### PR TITLE
Internal refactorings to install command.

### PR DESCRIPTION
**Pull request by m-renaud**
Pulls out implementation details into standalone functions. This makes the required inputs clear and removes from-afar data dependencies.

---
Please include the following checklist in your PR:

* [x] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#conventions).
* [x] Any changes that could be relevant to users have been recorded in the changelog (add file to `changelog.d` directory).
* [x] The documentation has been updated, if necessary.

Please also shortly describe how you tested your change. Bonus points for added tests!

### CODE REVIEW

File: cabal-install/Distribution/Client/CmdInstall.hs [hunk 1]
```diff
@@ -628,6 +464,164 @@ installAction ( configFlags, configExFlags, installFlags
                   haddockFlags testFlags benchmarkFlags
     globalConfigFlag = projectConfigConfigFile (projectConfigShared cliConfig)
 
+
+getClientInstallFlags :: Verbosity -> GlobalFlags -> ClientInstallFlags -> IO ClientInstallFlags
```
Comment (phadej) [inline cabal-install/Distribution/Client/CmdInstall.hs:468]: The name should hint that this is used for `ignore-project` cases.

File: cabal-install/Distribution/Client/CmdInstall.hs [hunk 2]
```diff
@@ -879,6 +886,72 @@ entriesForLibraryComponents = Map.foldrWithKey' (\k v -> mappend (go k v)) []
...
+  ".ghc.environment." <> ghcPlatformAndVersionString platform compilerVersion
+
+getPackageDbStack
+  :: CompilerId
+  -> Flag FilePath
+  -> Flag FilePath
+  -> IO PackageDBStack
+getPackageDbStack compilerId storeDirFlag logsDirFlag = do
+  cabalDir <- getCabalDir
+  mstoreDir <- sequenceA $ makeAbsolute <$> flagToMaybe storeDirFlag
```
Comment (phadej) [inline cabal-install/Distribution/Client/CmdInstall.hs:949]: `traverse`

File: cabal-install/Distribution/Client/CmdInstall.hs [hunk 3]
```diff
@@ -879,6 +886,72 @@ entriesForLibraryComponents = Map.foldrWithKey' (\k v -> mappend (go k v)) []
...
+      _                     -> False
+
+-- | Constructs the path to the global GHC environment file.
+getGlobalEnv :: FilePath -> Platform -> Version -> String -> FilePath
+getGlobalEnv home platform compilerVersion name =
+  home </> ".ghc" </> ghcPlatformAndVersionString platform compilerVersion
+  </> "environments" </> name
+
+-- | Constructs the path to a local GHC environment file.
+getLocalEnv :: Platform -> Version -> FilePath -> FilePath
```
Comment (phadej) [inline cabal-install/Distribution/Client/CmdInstall.hs:937]: I'd put `FilePath` as the first argument.

File: cabal-install/Distribution/Client/CmdInstall.hs [hunk 4]
```diff
@@ -879,6 +886,72 @@ entriesForLibraryComponents = Map.foldrWithKey' (\k v -> mappend (go k v)) []
...
+        " is unparsable. Libraries cannot be installed.") >> return []
+    else return []
+  where
+    -- Why? We know what the first part will be, we only care about the packages.
+    filterEnvEntries = filter $ \case
+      GhcEnvFilePackageId _ -> True
+      _                     -> False
+
+-- | Constructs the path to the global GHC environment file.
+getGlobalEnv :: FilePath -> Platform -> Version -> String -> FilePath
```
Comment (phadej) [inline cabal-install/Distribution/Client/CmdInstall.hs:931]: TODO for the future: make newtype for `PkgEnvName`.

File: cabal-install/Distribution/Client/CmdInstall.hs (hunk 1 above)
Comment (m-renaud) [reply cabal-install/Distribution/Client/CmdInstall.hs:468]: Hmm, I'm not familiar with `ignore-project` cases or what they imply so can't really come up with a better name, let me know if you have a suggestion.

File: cabal-install/Distribution/Client/CmdInstall.hs (hunk 2 above)
Comment (m-renaud) [reply cabal-install/Distribution/Client/CmdInstall.hs:949]: Done.

File: cabal-install/Distribution/Client/CmdInstall.hs (hunk 3 above)
Comment (m-renaud) [reply cabal-install/Distribution/Client/CmdInstall.hs:937]: Done.

File: cabal-install/Distribution/Client/CmdInstall.hs (hunk 4 above)
Comment (m-renaud) [reply cabal-install/Distribution/Client/CmdInstall.hs:931]: Done.

**Comment by phadej** [comment]
Please squash into single commit and merge once CI is green

**Pull request merged**